test/gtest/CMakeLists.txt

```diff
@@ -32,7 +32,9 @@ function(add_gtest TEST_NAME TEST_CPP)
   endif()
   if(MIOPEN_TEST_DISCRETE)
     # Enable CMake to discover the test binary
-    gtest_discover_tests(${TEST_NAME} DISCOVERY_TIMEOUT 300 DISCOVERY_MODE PRE_TEST WORKING_DIRECTORY ${PROJECT_BINARY_DIR}/${DATABASE_INSTALL_DIR} PROPERTIES ENVIRONMENT "MIOPEN_USER_DB_PATH=${CMAKE_CURRENT_BINARY_DIR};MIOPEN_TEST_FLOAT_ARG=${MIOPEN_TEST_FLOAT_ARG};MIOPEN_TEST_ALL=${MIOPEN_TEST_ALL};MIOPEN_TEST_MLIR=${MIOPEN_TEST_MLIR};MIOPEN_TEST_COMPOSABLEKERNEL=${MIOPEN_TEST_COMPOSABLEKERNEL};CODECOV_TEST=${CODECOV_TEST};MIOPEN_TEST_DBSYNC=${MIOPEN_TEST_DBSYNC};MIOPEN_TEST_CONV=${MIOPEN_TEST_CONV};MIOPEN_TEST_DEEPBENCH=${MIOPEN_TEST_DEEPBENCH};MIOPEN_DEBUG_TUNING_ITERATIONS_MAX=${MIOPEN_DEBUG_TUNING_ITERATIONS_MAX}")
+    # Note: Due to the following cmake issue with gtest_discover_tests https://gitlab.kitware.com/cmake/cmake/-/issues/17812 you cannot pass PROPERTIES as a list.
+    #       To work around this limitation, we are passing the environment variables in the format ENVIRONMENT;value1=${value1};ENVIRONMENT;value2=${value2}.
+    gtest_discover_tests(${TEST_NAME} DISCOVERY_TIMEOUT 300 DISCOVERY_MODE PRE_TEST WORKING_DIRECTORY ${PROJECT_BINARY_DIR}/${DATABASE_INSTALL_DIR} PROPERTIES "ENVIRONMENT;MIOPEN_USER_DB_PATH=${CMAKE_CURRENT_BINARY_DIR};ENVIRONMENT;MIOPEN_TEST_FLOAT_ARG=${MIOPEN_TEST_FLOAT_ARG};ENVIRONMENT;MIOPEN_TEST_ALL=${MIOPEN_TEST_ALL};ENVIRONMENT;MIOPEN_TEST_MLIR=${MIOPEN_TEST_MLIR};ENVIRONMENT;MIOPEN_TEST_COMPOSABLEKERNEL=${MIOPEN_TEST_COMPOSABLEKERNEL};ENVIRONMENT;CODECOV_TEST=${CODECOV_TEST};ENVIRONMENT;MIOPEN_TEST_DBSYNC=${MIOPEN_TEST_DBSYNC};ENVIRONMENT;MIOPEN_TEST_CONV=${MIOPEN_TEST_CONV};ENVIRONMENT;MIOPEN_TEST_DEEPBENCH=${MIOPEN_TEST_DEEPBENCH};ENVIRONMENT;MIOPEN_DEBUG_TUNING_ITERATIONS_MAX=${MIOPEN_DEBUG_TUNING_ITERATIONS_MAX};ENVIRONMENT;MIOPEN_TEST_WITH_MIOPENDRIVER=${MIOPEN_TEST_WITH_MIOPENDRIVER};ENVIRONMENT;MIOPENDRIVER_MODE_CONV=${MIOPENDRIVER_MODE_CONV};ENVIRONMENT;MIOPENDRIVER_MODE_GEMM=${MIOPENDRIVER_MODE_GEMM};ENVIRONMENT;MIOPENDRIVER_MODE_BN=${MIOPENDRIVER_MODE_BN};ENVIRONMENT;MIOPENDRIVER_MODE_POOL=${MIOPENDRIVER_MODE_POOL}")
```

Comment:
`MIOPENDRIVER_MODE_CONV`, `MIOPENDRIVER_MODE_GEMM`, `MIOPENDRIVER_MODE_BN`, `MIOPENDRIVER_MODE_POOL` should not be exported, these are intended for use only in CMakeLists.txt (just like macros in C, to have some single place where these command line options of the driver are defined).

gtests should have their own definitions. To be able to work standalone, at least.

Reply:
Ah good point.

Reply:
Updated, please check again when you get a chance!

I removed the env variables for MIOPENDRIVER_MODE_X, and updated the tests to be split to enable running them standalone properly.